routes/items.js

```diff
@@ -18,14 +25,29 @@ router.post("/", (req, res, next) => {
 });
 
 router.put("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  // complete writing this route handler
+  // res.status(501).send({ error: 'route not yet implemented' });
+  const itemUpdated = itemData.getById(req.params.id);
```

Comment:
The logic of whether there is an item associated with the sent id should remain in the data interface method. Here in the route handler, only a call to itemData.updateById should be made. We want route handlers to remain as ignorant as possible about data related business logic. This would be better, as it makes only a single call to the data interface and returns the updated item not the original version of it: 
```
router.put("/:id", (req, res, next) => {
    const itemUpdated =    itemData.updateById(req.params.id, req.body);
    if(itemUpdated){ 
      res.status(200).send(itemUpdated);
    } else {
      res.sendStatus(400)
    }
```